pydicom/charset.py

```diff
@@ -546,40 +572,47 @@ def _encode_string_parts(value, encodings):
                 best_encoding = encoding
                 max_index = len(unencoded_part)
                 break
-            except UnicodeError as e:
-                if e.start > max_index:
-                    # e.start is the index of first character failed to encode
-                    max_index = e.start
+            except UnicodeEncodeError as err:
```

Comment:
Needs to be narrower to pass typing

Reply:
I dimly remember that I did this for a reason... if think `encode` could also raise `UnicodeDecodeError` (was quite unexpected to me), but maybe that was a Python 2 thing.

Reply:
I'll put `UnicodeDecodeError` in just in case